src/backend/storage/lmgr/lock.c

```diff
@@ -3545,7 +3552,7 @@ GetLockStatusData(void)
 			instance->distribXid = (Gp_role == GP_ROLE_DISPATCH)?
 								   proc->gxact.gxid :
 								   proc->localDistribXactData.distribXid;
-			instance->holdTillEndXact = false;
+			instance->holdTillEndXact = proc->fpHoldTillEndXact[f];
```

Comment:
I don't think the logic is right here. One `instance` for fast-path stands for at most 3 locks. So which lock `holdTillEndXact` is describing for?

Reply:
@guofengrichard I understand slots wrong.

Reply:
@guofengrichard 

As you mentioned:
```
One instance for fast-path stands for at most 3 locks. So which lock holdTillEndXact is describing for?
```

And instance->holdTillEndXact is a bool. How can a bool describe 3 locks' status?